src/javafundamentals/arrays/ProjectTicTacToe.java

```diff
@@ -1,16 +1,200 @@
+
+        if (checkWin("X")) {
+            printBoard();
+            System.out.println("Congratulations, you won!!");
+        } else if (checkDraw()) {
+            printBoard();
+            System.out.println("It's a draw!");
+        } else {
+
+            aiSelection();
+
+            printBoard();
+
+            if (checkWin("O")) {
+                printBoard();
+                System.out.println("Sorry, you lost this game...");
+            } else {
+                playGame();
+            }
+        }
+    }
+
+
+    private static void userSelection() {
+        Scanner scanner = new Scanner(System.in);
+        try {
+            String spotChosen = scanner.nextLine();
+            spotChosen.toLowerCase();
+            boolean validSpotChosen = false;
+
+            choiceChecker:
+            for (int i = 0; i < gameBoard.length; i++) {
+                for (int j = 0; j < gameBoard[i].length; j++) {
+                    if (gameBoard[i][j].equals(spotChosen)) {
+                        gameBoard[i][j] = ("X");
+                        validSpotChosen = true;
+                        break choiceChecker;
+                    }
+                }
+            }
+            if (!validSpotChosen) {
+                System.out.println("This spot cannot be chosen, try again");
+                userSelection();
+            }
+
+        } catch (Exception e) {
+            System.out.println("Invalid input, try again");
+            userSelection();
+        }
+    }
+
+    private static void aiSelection() {
+        int randomColumn = (int) (Math.random() * gameBoard.length + 0);
+        int randomrow = (int) (Math.random() * gameBoard.length + 0);
+
+        if (!gameBoard[randomColumn][randomrow].equals("X") && !gameBoard[randomColumn][randomrow].equals("O")) {
+            gameBoard[randomColumn][randomrow] = "O";
+        } else {
+            aiSelection();
+        }
+    }
+
+    private static boolean checkWin(String letter) {
+        boolean winsHorizontally = false;
+
+        //Check horizontal wins
+        for (int i = 0; i < gameBoard.length; i++) {
+            winsHorizontally = true;
+            for (int j = 0; j < gameBoard[i].length; j++) {
+                winsHorizontally = winsHorizontally && gameBoard[i][j].equals(letter);
+            }
+            if (winsHorizontally) {
+                return true;
+            }
+        }
+
+        boolean winsVertically = false;
+
+        for (int i = 0; i < gameBoard.length; i++) {
+            winsVertically = true;
+            for (int j = 0; j < gameBoard[i].length; j++) {
+                winsVertically = winsVertically && gameBoard[j][i].equals(letter);
+            }
+            if (winsVertically) {
+                return true;
+            }
+        }
+
+        boolean winsDiagonallyRight = true;
+
+        for (int i = 0; i < gameBoard.length; i++) {
+            winsDiagonallyRight = winsDiagonallyRight && gameBoard[i][i].equals(letter);
```

Comment:
This is a good way of checking for a win, but if you turn the 3x3 tic tac toe into a 4x4 then it wont work for the positions that are not in the center diagonal